tools-v2/pkg/cli/command/curvebs/update/leader_schedule/leader_schedule.go

```diff
@@ -0,0 +1,165 @@
+ */
+
+/*
+* Project: curvecli
+* Created Date: 2023-06-14
+* Author: montaguelhz
+ */
+
+package leader_schedule
+
+import (
+	"context"
+	"fmt"
+
+	cmderror "github.com/opencurve/curve/tools-v2/internal/error"
+	cobrautil "github.com/opencurve/curve/tools-v2/internal/utils"
+	basecmd "github.com/opencurve/curve/tools-v2/pkg/cli/command"
+	"github.com/opencurve/curve/tools-v2/pkg/config"
+	"github.com/opencurve/curve/tools-v2/pkg/output"
+	"github.com/opencurve/curve/tools-v2/proto/proto/schedule"
+	"github.com/opencurve/curve/tools-v2/proto/proto/schedule/statuscode"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc"
+)
+
+type UpdateLeaderScheduleRpc struct {
+	Info    *basecmd.Rpc
+	Request *schedule.RapidLeaderScheduleRequst
+	Client  schedule.ScheduleServiceClient
+}
+
+var _ basecmd.RpcFunc = (*UpdateLeaderScheduleRpc)(nil) // check interface
+
+func (uRpc *UpdateLeaderScheduleRpc) NewRpcClient(cc grpc.ClientConnInterface) {
+	uRpc.Client = schedule.NewScheduleServiceClient(cc)
+}
+
+func (uRpc *UpdateLeaderScheduleRpc) Stub_Func(ctx context.Context) (interface{}, error) {
+	return uRpc.Client.RapidLeaderSchedule(ctx, uRpc.Request)
+}
+
+type LeaderScheduleCommand struct {
+	basecmd.FinalCurveCmd
+
+	Rpc *UpdateLeaderScheduleRpc
+}
+
+var _ basecmd.FinalCurveCmdFunc = (*LeaderScheduleCommand)(nil) // check interface
+
+const (
+	leaderScheduleExample = `$ curve bs update leader-schedule --logicalpoolid 1
+$ curve bs update leader-schedule --all`
+)
+
+func NewLeaderScheduleCommand() *cobra.Command {
+	return NewUpdateLeaderScheduleCommand().Cmd
+}
+
+func (lCmd *LeaderScheduleCommand) AddFlags() {
+	config.AddBsMdsFlagOption(lCmd.Cmd)
+	config.AddRpcRetryTimesFlag(lCmd.Cmd)
+	config.AddRpcTimeoutFlag(lCmd.Cmd)
+
+	config.AddBsLogicalPoolIdOptionFlag(lCmd.Cmd)
+	config.AddBsAllOptionFlag(lCmd.Cmd)
+}
+
+func NewUpdateLeaderScheduleCommand() *LeaderScheduleCommand {
+	sCmd := &LeaderScheduleCommand{
+		FinalCurveCmd: basecmd.FinalCurveCmd{
+			Use:     "leader-schedule",
+			Short:   "rapidly transfer leader",
+			Example: leaderScheduleExample,
+		},
+	}
+
+	basecmd.NewFinalCurveCli(&sCmd.FinalCurveCmd, sCmd)
+	return sCmd
+}
+
+func (lCmd *LeaderScheduleCommand) Init(cmd *cobra.Command, args []string) error {
+	mdsAddrs, err := config.GetBsMdsAddrSlice(lCmd.Cmd)
+	if err.TypeCode() != cmderror.CODE_SUCCESS {
+		return err.ToError()
+	}
+
+	header := []string{cobrautil.ROW_RESULT, cobrautil.ROW_REASON}
+	lCmd.SetHeader(header)
+
+	timeout := config.GetFlagDuration(lCmd.Cmd, config.RPCTIMEOUT)
+	retryTimes := config.GetFlagInt32(lCmd.Cmd, config.RPCRETRYTIMES)
+
+	if !lCmd.Cmd.Flag(config.CURVEBS_ALL).Changed && !lCmd.Cmd.Flag(config.CURVEBS_LOGIC_POOL_ID).Changed {
+		return fmt.Errorf("all or logicalpoolid is required")
+	}
+
+	all := config.GetBsFlagBool(lCmd.Cmd, config.CURVEBS_ALL)
+	if !all && !lCmd.Cmd.Flag(config.CURVEBS_LOGIC_POOL_ID).Changed {
+		return fmt.Errorf("if set flag all false, logicalpoolid is required")
```

Comment:
```suggestion
		return fmt.Errorf("all or logicalpoolid is required")
```

Reply:
fine, I found the judgement is unnecessary.